sys/include/net/sock/udp.h

```diff
@@ -13,6 +13,245 @@
  * @defgroup    net_sock_udp    UDP sock API
  * @ingroup     net_sock
  * @brief       Sock submodule for UDP
+ *
+ * How To Use
+ * ----------
+ * First you need to @ref including-modules "include" a module that implements
+ * this API in your application's Makefile. For example the implementation for
+ * @ref net_gnrc "GNRC" is called `gnrc_sock_udp`.
+ *
+ * ### A Simple UDP Echo Server
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include <stdio.h>
+ *
+ * #include "net/af.h"
+ * #include "net/sock/udp.h"
+ *
+ * uint8_t buf[128];
+ *
+ * int main(void)
+ * {
+ *     sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ *     sock_udp_t sock;
+ *
+ *     local.port = 12345;
+ *
+ *     if (sock_udp_create(&sock, &local, NULL, 0) < 0) {
+ *         puts("Error creating UDP sock");
+ *         return 1;
+ *     }
+ *
+ *     while (1) {
+ *         sock_udp_ep_t remote;
+ *         ssize_t res;
+ *
+ *         if ((res = sock_udp_recv(&sock, buf, sizeof(buf), 0, &remote)) >= 0) {
+ *             puts("Received a message");
+ *             if (sock_udp_send(&sock, buf, res, &remote) < 0) {
+ *                 puts("Error sending reply");
+ *             }
+ *         }
+ *     }
+ *
+ *     return 0;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Above you see a simple UDP echo server. Don't forget to also
+ * @ref including-modules "include" the IPv6 module of your networking
+ * implementation (e.g. gnrc_ipv6_default` for @ref net_gnrc GNRC) and at least
+ * one network device.
+ *
+ * After including header files for the @ref net_af "address families" and
+ * the @ref net_sock_ip "raw `sock`s" themselves, we create some buffer space
+ * `buf` to store the data received by the server:
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include "net/af.h"
+ * #include "net/sock/ip.h"
+ *
+ * uint8_t buf[128];
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * To be able to listen for incoming packets we want to bind the sock so we need
```

Comment:
we want to bind the sock so we need, etc
=> we bind the sock by setting a local end point with a port ?